src/core/execute.c

```diff
@@ -3606,11 +3613,9 @@ static int add_shifted_fd(int *fds, size_t fds_size, size_t *n_fds, int fd, int
                 return 0;
         }
 
-        if (fd < 3 + (int) *n_fds) {
-                /* Let's move the fd up, so that it's outside of the fd range we will use to store
-                 * the fds we pass to the process (or which are closed only during execve). */
+        if (fd < threshold) {
 
-                r = fcntl(fd, F_DUPFD_CLOEXEC, 3 + (int) *n_fds);
+                r = fcntl(fd, F_DUPFD_CLOEXEC, threshold);
```

Comment:
maybe mention in a comment, that this doesn't mean it's actually going to be 255, but just something >= 255, in case the fd store is crowded or users define gazillions of .socket units for the same service